### PR TITLE
seperate indirect plot preview model

### LINKED ISSUE #29210: Improve comunication between gui parts in indirect data analysis
The poor comunication between parts of the gui makes it very easy for small changes to the indirect gui to spiral out and cause more bugs. the comunication between the gui parts needs to be recactored almost entirly.

One part of this or improve comunication is to use a single class containing the workspaces that each of the portions can be aware of and comunicate with, like @Mantid-Matthew has begun doing.

### PR DESCRIPTION
**Pull request by Harrietbrown**
**Description of work.**

This PR separates aspects of the IndirectFitPlotModel from the IndirectFittingModel and clears remaining connections to IndirectFitDataModel.

**To test:**

There should be no changes to fucntionality with this PR. It is important to check that the standard usage of all the tabs opperate correctly.

Fixes #29210 

This does not require release notes because It is entirly internal changes

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
@Harrietbrown is this for 6.2 or 6.3?

**Comment by Harrietbrown** [comment]
6.3, I forgot to add the milestone.

File: qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp [hunk 1]
```diff
@@ -99,6 +99,9 @@ void IndirectFitAnalysisTab::connectDataPresenter() {
 void IndirectFitAnalysisTab::connectPlotPresenter() {
   connect(m_plotPresenter.get(), SIGNAL(fitSingleSpectrum(WorkspaceID, WorkspaceIndex)), this,
           SLOT(singleFit(WorkspaceID, WorkspaceIndex)));
+  connect(m_plotPresenter.get(), SIGNAL(startXChanged(double)), this, SLOT(startXChanged(double)));
+  connect(m_plotPresenter.get(), SIGNAL(endXChanged(double)), this, SLOT(endXChanged(double)));
```
Comment (sf1919) [inline qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp:103]: These seem to have the same calls for both the Signal and the Slot - could this get confusing?

File: qt/scientific_interfaces/Indirect/IndirectFitData.cpp [hunk 2]
```diff
@@ -282,12 +282,17 @@ void IndirectFitData::validateSpectra(FunctionModelSpectra const &spectra) {
 
 void IndirectFitData::setStartX(double const &startX, WorkspaceIndex const &spectrum) {
   const auto range = m_ranges.find(spectrum);
-  if (range != m_ranges.end())
-    range->second.first = startX;
-  else if (m_workspace)
+  if (range != m_ranges.end()) {
+    if (range->second.second < startX) {
+      range->second.first = range->second.second;
```
Comment (sf1919) [inline qt/scientific_interfaces/Indirect/IndirectFitData.cpp:287]: Can you tell me a bit more about these two lines? what are second.first and second.second in range?

File: qt/scientific_interfaces/Indirect/IndirectFitOutput.cpp [hunk 3]
```diff
@@ -6,7 +6,7 @@
 // SPDX - License - Identifier: GPL - 3.0 +
 #include <memory>
 
-#include "IndirectFitOutputModel.h"
+#include "IndirectFitOutput.h"
```
Comment (sf1919) [inline qt/scientific_interfaces/Indirect/IndirectFitOutput.cpp:9]: How does the IndirectFitOutput work in terms of MVP? It this just a rename and this still acts as a Model or does this have a new function now?

File: qt/scientific_interfaces/Indirect/IndirectFitPlotPresenter.h [hunk 4]
```diff
@@ -9,6 +9,7 @@
 #include "DllConfig.h"
 
 #include "IndirectFitPlotModel.h"
+#include "MantidAPI/IFunction_fwd.h"
```
Comment (sf1919) [inline qt/scientific_interfaces/Indirect/IndirectFitPlotPresenter.h:12]: What is this included for?

File: qt/scientific_interfaces/Indirect/IndirectFitData.cpp (hunk 2 above)
Comment (Harrietbrown) [reply qt/scientific_interfaces/Indirect/IndirectFitData.cpp:287]: range here is an itterator for a map, `range.first` is a WorkspaceIndex, and `range.second` is a pair of doubles where the `range.second.first` is StartX and `range.second.second` is EndX

File: qt/scientific_interfaces/Indirect/IndirectFitOutput.cpp (hunk 3 above)
Comment (Harrietbrown) [reply qt/scientific_interfaces/Indirect/IndirectFitOutput.cpp:9]: `IndirectFitOutput` is an object for storing and accessing results from fits in the same way we use `IndirectFitData` to store input data.

File: qt/scientific_interfaces/Indirect/IndirectFitPlotPresenter.h (hunk 4 above)
Comment (Harrietbrown) [reply qt/scientific_interfaces/Indirect/IndirectFitPlotPresenter.h:12]: Must have been left over, I'll remove it

Comment (sf1919) [reply qt/scientific_interfaces/Indirect/IndirectFitPlotPresenter.h:12]: I also spotted it in IndirectFitPlotModel.h file.

File: qt/scientific_interfaces/Indirect/IndirectFitOutput.cpp (hunk 3 above)
Comment (sf1919) [reply qt/scientific_interfaces/Indirect/IndirectFitOutput.cpp:9]: Is it worth explaining this in the Dev Docs about the Indirect File Structure?

Comment (Harrietbrown) [reply qt/scientific_interfaces/Indirect/IndirectFitOutput.cpp:9]: Is it not clear in the current dev-docs:

```The IndirectFitPlotModel also contains a pointer to the active fit function and the IndirectFitOutput from the
IndirectFittingModel, again it should only ever READ FROM THESE OBJECTS only the IndirectFittingModel should control it.```

File: qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp (hunk 1 above)
Comment (sf1919) [reply qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp]: Thanks for renaming these

File: dev-docs/source/IndirectDataAnalysisFileStructure.rst [hunk 5]
```diff
@@ -3,81 +3,32 @@
 IndirectDataAnalysis File Structure
 ===================================
 
-The many levels of inheritance in the indirect data analysis codebase can be confusing.
+The many levels of inheritance in the indirect data analysis codebase can be confusing. it is not always clear how they
+all interact.
 
-Tabs
-----
+The Ewlin and I(Q,t) are the most different as their components are all contained within a single UI file. This is
```
Comment (sf1919) [inline dev-docs/source/IndirectDataAnalysisFileStructure.rst:9]: Should this be Elwin?

**Review by sf1919 — APPROVED** [review]
Assuming that all the tests pass....

I have tested this and all the file loads/workspace loads for Indirect Data Analysis still work. All my earlier comments have been responded to and where necessary the code updated. 

The structure of Indirect Data Analysis is now much easier to navigate for developers. There is still some work to do (around Elwin in particular) but this is a definite improvement.

**Pull request merged**